node/ddos.go

```diff
@@ -0,0 +1,166 @@
+package node
+
+import (
+	"sync"
+	"time"
+
+	"github.com/harmony-one/abool"
+	"github.com/harmony-one/harmony/p2p"
+	libp2p_peer "github.com/libp2p/go-libp2p-core/peer"
+	libp2p_pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"golang.org/x/time/rate"
+)
+
+const (
+	pubSubRateLimit  = 5  // 5 messages per second
+	pubSubBurstLimit = 20 // 20 messages at burst
+
+	// When a node is bootstrapped, it will be flooded with some pub-sub message from the past.
+	// Ease the rate limit at the initial bootstrap.
+	rateLimiterEasyPeriod = 5 * time.Second
+)
+
+// Preconfigured internal nodes
+// TODO: replace all internal node peer ID here
+var trustedPeers = []string{
```

Comment:
no guarantee the trustedPeerID won't change, they are generated randomly.  If we change the validator nodes, they will be different.

did you consider all the bootnodes, explorer nodes, snapshot nodes?